Update stm32f4 SDIO register fields and values

jkristell opened the pull request:
Documentation and limits for the sdio register values.

jkristell commented:
I only added it to the f405 for now. But if this change looks OK to you, I can add the sdio peripheral to all other f4 devices as well. Just wanted to make sure it was the right approach.

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -0,0 +1,113 @@
+    DATATIME: [0, 0xFFFFFFFF]
+
+  DLEN:
+    DATALENGTH: [0, 0xFFFFFF]
```
```suggestion
    DATALENGTH: [0, 0x01FFFFFF]
```

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -0,0 +1,113 @@
+    RWSTART:
+      Disabled: [0, "Don't start read wait operation"]
+      Enabled: [1, "Read wait operation starts"]
+    DBLOCKSIZE: [0, 0xFF]
```
```suggestion
    DBLOCKSIZE: [0, 0xF]
```

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -0,0 +1,113 @@
+      Enabled: [1, "Start transfer"]
+
+  DCOUNT:
+    DATACOUNT: [0, 0x00FFFFFF]
```
```suggestion
    DATACOUNT: [0, 0x01FFFFFF]
```

jkristell commented:
Thank you for the review, and sorry for the field length errors, I should have double checked them. Will do that next time!

I updated the branch with your fixes and ran matchperipheral.py and let it add the peripheral to the devieces and added the FIFOData field to the SDIO_FIFO.

I didn't plan to bother with the interrupt register fields. As they are all bit fields, I didn't think it was worth the effort. But I can be convinced to do a follow up patch.

adamgreig commented:
Great, thanks!

> I didn't plan to bother with the interrupt register fields. As they are all bit fields, I didn't think it was worth the effort. But I can be convinced to do a follow up patch.

Since they are all simple fields you might be able to do something very simple like:

```yaml
SDIO_MASK:
    "*IE":
        Disabled: [0, "Interrupt disabled"]
        Enabled: [1, "Interrupt enabled"]
SDIO_ICR:
    "*C":
         _write:
              Clear: [1, "Clear flag"]
```

I think it might be worth doing the STA register fully though as I expect a lot of driver code will regularly be polling that register so it's much nicer if they can use the enumerated values (e.g. rxdavl().data_available() rather than rxdavl().bit_set()).

jkristell commented:
Will try to find some time to look at this later this week. So lets keep this open for the time being.

jkristell commented:
Updated with interrupt field value documentation.

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -106,6 +106,90 @@ SDIO:
 
+  STA:
+    CEATAEND:
+      NotReceiver: [0, "Completion signal not received"]
```
```suggestion
      NotReceived: [0, "Completion signal not received"]
```

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -106,6 +106,90 @@ SDIO:
+      NotHalfFull: [0, "Receive FIFO not half full"]
+      HalfFull: [1, "Receive FIFO half full. At least 8 words in the FIFO"]
+    TXFIFOHE:
+      NotHalfFull: [0, "Transmit FIFO not half full"]
+      HalfFull: [1, "Transmit FIFO half empty. At least 8 words can be written into the FIFO"]
```
```suggestion
      NotHalfEmpty: [0, "Transmit FIFO not half empty"]
      HalfEmpty: [1, "Transmit FIFO half empty. At least 8 words can be written into the FIFO"]
```

adamgreig commented on peripherals/sdio/sdio_f4.yaml:
```diff
@@ -106,6 +106,90 @@ SDIO:
+      NotInProgress: [0, "Command transfer not in progress"]
+      InProgress: [1, "Command tranfer in progress"]
+    DBCKEND:
+      NotTransfered: [0, "Data block not sent/received (CRC check failed)"]
+      Transfered: [1, "Data block sent/received (CRC check passed)"]
```
```suggestion
      NotTransferred: [0, "Data block not sent/received (CRC check failed)"]
      Transferred: [1, "Data block sent/received (CRC check passed)"]
```
